burr/core/application.py

```diff
@@ -2484,3 +2589,51 @@ def build(self) -> Application[StateType]:
             state_persister=self.state_persister,
             state_initializer=self.state_initializer,
         )
+
+    @telemetry.capture_function_usage
+    def build(self) -> Application[StateType]:
+        """Builds the application.
+
+        This function is a bit messy as we iron out the exact logic and rigor we want around things.
+
+        :return: The application object
+        """
+        _validate_app_id(self.app_id)
+        if self.state is None:
+            self.state = State()
+
+        if self.state_persister:
+            self._set_sync_state_persister()  # this is used to save the state during application run
+        if self.state_initializer:
+            # sets state, sequence_id, and maybe start
+            self._load_from_sync_persister()  # this is used to load application from a previously saved state
+
+        return self._build_common()
+
+    @telemetry.capture_function_usage
+    async def abuild(self) -> Application[StateType]:
+        """Builds the application.
+
+        This function is a bit messy as we iron out the exact logic and rigor we want around things.
```

Comment:
No longer messy, fix this. And make it + `build()` clear about the parameters (good place to have this table?).